packages/mui-material/src/Chip/Chip.js

```diff
@@ -156,141 +158,190 @@ const ChipRoot = styled('div', {
+                    : alpha(theme.palette[color].contrastText, 0.7),
+                  '&:hover, &:active': {
+                    color: (theme.vars || theme).palette[color].contrastText,
+                  },
+                },
+                [`&.${chipClasses.focusVisible}.${chipClasses.deletable}`]: {
+                  background: (theme.vars || theme).palette[color].dark,
+                },
+              },
+            };
+          }),
+        {
+          props: isIconColorEqualToColor,
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: theme.vars ? theme.vars.palette.Chip.defaultIconColor : textColor,
+            },
+          },
+        },
+        {
+          props: (props) => isIconColorEqualToColor(props) && props.color !== 'default',
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: 'inherit',
+            },
+          },
+        },
+        {
+          props: (props) => !!props.onDelete,
+          style: {
+            [`&.${chipClasses.focusVisible}`]: {
+              backgroundColor: theme.vars
+                ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
+                : alpha(
+                    theme.palette.action.selected,
+                    theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
+                  ),
+            },
           },
-        }),
+        },
+      ],
     };
   },
-  ({ theme, ownerState }) => ({
-    ...(ownerState.clickable && {
-      userSelect: 'none',
-      WebkitTapHighlightColor: 'transparent',
-      cursor: 'pointer',
-      '&:hover': {
-        backgroundColor: theme.vars
-          ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.hoverOpacity}))`
-          : alpha(
-              theme.palette.action.selected,
-              theme.palette.action.selectedOpacity + theme.palette.action.hoverOpacity,
-            ),
-      },
-      [`&.${chipClasses.focusVisible}`]: {
-        backgroundColor: theme.vars
-          ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
-          : alpha(
-              theme.palette.action.selected,
-              theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
-            ),
-      },
-      '&:active': {
-        boxShadow: (theme.vars || theme).shadows[1],
-      },
-    }),
-    ...(ownerState.clickable &&
-      ownerState.color !== 'default' && {
-        [`&:hover, &.${chipClasses.focusVisible}`]: {
-          backgroundColor: (theme.vars || theme).palette[ownerState.color].dark,
+  ({ theme }) => ({
+    variants: [
+      {
+        props: isClickable,
```

Comment:
For readability, if `isClickable` is not reused, the function can be inlined here I think.

Reply:
It wasn't needed in the end